### ISSUE TITLE
Update from a release on a branch (question)

### ISSUE THREAD

**Issue by eric-gagnon** [YahnisElsts/plugin-update-checker#191]
Hello,

I'm not sure if I use the update the right way.

I intend to use trunk as a ongoing developpement branch and do a branch for release only if I have to handle hot fixes, like that : https://barro.github.io/2016/02/a-succesful-git-branching-model-considered-harmful/.

I tried to reproduce that setup but I have the impression that the release / tag is not discovered appropriately. I have to change the code to specify the branch so that the update is notified
$update_checker->setBranch('test-mise-a-jour');

Obviously, I cannot use that as the release branch will not be known in advance.

Any idea of what I'm doing wrong or how I could handle such scenario? (I need to use release to handle the build of vendor folder). I will try to trace the code on my side and see if I can understand what's going on...

**Comment by eric-gagnon** [comment]
Already, I can see that the plugin file is not resolved here (but the release 1.0.1 info is good).

	$mainPluginFile = basename($this->pluginFile);
			$remotePlugin = $api->getRemoteFile($mainPluginFile, $ref);
			if ( !empty($remotePlugin) ) {
				$remoteHeader = $this->getFileHeader($remotePlugin);
				$this->setInfoFromHeader($remoteHeader, $info);
			}

**Comment by YahnisElsts** [comment]
Assuming that I've understood your approach, I think that tags would still work and the update checker would be able to discover them. If you create a tag "v2.0" on the branch "release-whatever-xyz", the update checker should detect that tag as a new version.

Therefore, I would recommend leaving the branch setting to the default value ("master") and using tags.

**Comment by eric-gagnon** [comment]
Yeah ok, I think I got it right, I checked the value of remote file and it was pointing to master, I probably forgot to change the branch when I did the release. I redid the release and it's working as expected now. I would probably help validate that if Github was displaying branch info more explicitely beside the release.

Thanks for confirming that it's legit use.

![screen shot 03-12-18 at 11 16 am](https://user-images.githubusercontent.com/21067311/37292360-2c44629a-25e7-11e8-9134-07e23131c421.JPG)

I close.
